src/control/tools/EditSelection.cpp

```diff
@@ -46,8 +46,9 @@ EditSelection::EditSelection(UndoRedoHandler* undo, Selection* selection, XojPag
 
 	for (Element* e : selection->selectedElements)
 	{
+		int indexWithinSourceLayer = this->sourceLayer->indexOf(e);
 		this->sourceLayer->removeElement(e, false);
-		addElement(e);
+		addElement(e, indexWithinSourceLayer);
```

Comment:
> You should/can also just inline this variable, since it is only used once.
I don't think this variable can be inlined because `removeElement` will change the returned index to -1.

Reply:
yes, you are right, but in the current version it is ordered like it has to be :)